devices/At24cxx/At24Base.cs

```diff
@@ -0,0 +1,185 @@
+        }
+
+        /// <summary>
+        /// Helper function to verify a memory address falls inside the accessible memory range for a At24c device.
+        /// </summary>
+        /// <param name="address">The memory address.</param>
+        /// <param name="availableMemory">The amount of memory available on the device.</param>
+        private static void VerifyMemoryAddress(int address, int availableMemory)
+        {
+            if (address < 0 || address > (availableMemory - 1))
+            {
+                throw new ArgumentOutOfRangeException(nameof(address), "Address must be within addressable memory range.");
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="At24Base" /> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to use for communication.</param>
+        /// <param name="pageSize">The device page size in bytes.</param>
+        /// <param name="pageCount">The number of pages on the device.</param>
+        public At24Base(I2cDevice i2cDevice, ushort pageSize, ushort pageCount)
+        {
+            if (i2cDevice == null)
+            {
+                throw new ArgumentNullException(nameof(i2cDevice));
+            }
+
+            _i2cDevice = i2cDevice;
+            PageSize = pageSize;
+            PageCount = pageCount;
+            Size = pageSize * pageCount;
+        }
+
+        /// <summary>
+        /// Reads a single byte from the device at the address following the last byte read or written.
+        /// </summary>
+        /// <returns>The byte read from the device.</returns>
+        public byte ReadByte()
+        {
+            return _i2cDevice.ReadByte();
+        }
+
+        /// <summary>
+        /// Reads a single byte from the device at the given address.
+        /// </summary>
+        /// <param name="address">The address to read from.</param>
+        /// <returns>The byte read from the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public byte ReadByte(int address)
+        {
+            VerifyMemoryAddress(address, Size);
+
+            byte[] readBuffer = new byte[1];
+            byte[] writeBuffer = CreateWriteBuffer(address, 0);
+
+            _i2cDevice.WriteRead(writeBuffer, readBuffer);
+
+            return readBuffer[0];
+        }
+
+        /// <summary>
+        /// Reads multiple bytes from the device starting from the given address.
+        /// </summary>
+        /// <param name="address">The address to read from.</param>
+        /// <param name="length">The number of bytes to read.</param>
+        /// <returns>The bytes read from the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public byte[] Read(int address, int length)
```

Comment:
same here for address. Is it the same for the length?
Also any check for min/max?

Reply:
Ah, yes I did overlook the min/max check. Fixed.